03.bowling/bowling.rb

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+score = ARGV[0]
+
+scores = score.split(',')
+
+shots = []
+scores.each do |s|
+  if s == 'X'
+    shots << 10
+    shots << 0
+  else
+    shots << s.to_i
+  end
+end
+
+frames = []
+shots.each_slice(2) do |shot|
+  frames << shot
+end
+
+point = 0
+frames.each.with_index do |frame, i|
+  next_frame = i + 1
```

Comment:
`i + 1`で取れるのはフレームのデータそのものではなく、ただの添え字（数字）ですね。
であれば`next_frame_index`のような名前の方がいいと思うのですが、そもそも添え字ではなくフレームデータを以下のように代入した方がコードがスッキリしそうです。

```suggestion
  next_frame = frame[i + 1]
```

Reply:
ご指摘ありがとうございます🙇‍♂️
2点、確認しました。
- `i + 1`で取れるのはただの添え字なので、私の最初のコードの変数の命名は不親切。
- フレームデータ（個々の配列）をそもそも代入していった方がコードがすっきりする。

後者の形でループ内の処理を修正しました。https://github.com/junohm410/ruby-practices/pull/3/commits/273d91750cf31deed971df29057ce51af4e3e2a1